src/liballoc/borrow.rs

```diff
@@ -479,3 +479,15 @@ impl<'a> AddAssign<Cow<'a, str>> for Cow<'a, str> {
         }
     }
 }
+
+#[stable(feature = "cow_str_add_assign_char", since = "1.41.0")]
+impl AddAssign<char> for Cow<'_, str> {
+    fn add_assign(&mut self, rhs: char) {
+        if let Cow::Borrowed(lhs) = *self {
+            let mut s = String::with_capacity(lhs.len() + rhs.len_utf8());
+            s.push_str(lhs);
+            *self = Cow::Owned(s);
+        }
+        self.to_mut().push(rhs);
+    }
+}
```

Comment:
Please set your editor to end every line with a newline character, otherwise this is not a valid text file and will not pass CI. (`./x.py test src/tool/tidy`)

Reply:
Added a new line at the end.